Sources/OpenFeature/OpenFeatureAPI.swift

```diff
@@ -43,15 +58,28 @@ public class OpenFeatureAPI {
     }
 
     public func setEvaluationContext(evaluationContext: EvaluationContext) {
-        let oldContext = self._context
-        self._context = evaluationContext
-        getProvider()?.onContextSet(oldContext: oldContext, newContext: evaluationContext)
+        providerStatus = .reconciling
+        eventHandler.send(.reconciling)
+        do {
+            let oldContext = self._context
+            self._context = evaluationContext
+            try getProvider()?.onContextSet(oldContext: oldContext, newContext: evaluationContext)
+            providerStatus = .ready
+            eventHandler.send(.contextChanged)
```

Comment:
**What "Resolution Reason" do we return in this case?** 
From the [Spec](https://openfeature.dev/specification/types#resolution-details), `STALE` is the better fit. In this case, `STALE` would signify either a change in the evaluation rules in the backend or a change of evaluation context in the client.

**Who is responsible to set `STALE` as Reason?** 
Ideally the SDK should do it to keep the behaviour consistent across Providers. Note: setting/reading a Provider State and a Provider EvaluationContext needs to be coupled and atomic to avoid race conditions.